libraries/plugins/apis/database_api/database_api.cpp

```diff
@@ -1425,40 +1425,94 @@ DEFINE_API_IMPL( database_api_impl, verify_signatures )
 
 DEFINE_API_IMPL( database_api_impl, find_transaction )
 {
-    // Yet to check the argument list
-    // FC_ASSERT ( args.size() == 1 || args.size() == 2, "Expected 1-2 arguments, was ${n}", ("n", args.size()) );
-    // auto trx_id = { args[0].as< transaction_id_type >() };
-    // auto expiration_time = { args[1].as< time_point_sec >() };
+    FC_ASSERT ( args.size() == 1 || args.size() == 2, "Expected 1-2 arguments, was ${n}", ("n", args.size()) );
```

Comment:
This is not how arguments in `database_api` work. This was copied from `condenser_api`, which is an old API format that parsed args as a json array. Now args are passed as a struct. Please see other `database_api` methods as an example of what to do.